components/brave_talk/browser/brave_talk_frame_host.cc

```diff
@@ -0,0 +1,73 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_talk/browser/brave_talk_frame_host.h"
+
+#include <memory>
+#include <sstream>
+#include <string>
+#include <utility>
+
+#include "absl/types/optional.h"
+#include "base/bind.h"
+#include "base/callback_forward.h"
+#include "base/logging.h"
+#include "brave/browser/brave_talk/brave_talk_service.h"
+#include "brave/components/brave_talk/common/features.h"
+#include "build/build_config.h"
+#include "chrome/browser/bad_message.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+#include "content/browser/bad_message.h"
+#include "content/browser/renderer_host/frame_tree_node.h"
+#include "content/browser/renderer_host/navigation_controller_impl.h"
+#include "content/browser/renderer_host/render_frame_host_impl.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/render_frame_host.h"
+#include "content/public/browser/web_contents.h"
+#include "third_party/blink/public/common/tokens/tokens.h"
+
+namespace brave_talk {
+
+namespace {
+content::RenderFrameHost* FindFrameForToken(
+    const absl::optional<::blink::FrameToken>& frame_token,
+    content::WebContents* contents) {
+  auto* main_frame =
```

Comment:
@simonhong would you mind taking a look at in particular. I feel like this is probably a layering violation, but I also couldn't find an API for getting a RenderFrameHost from a FrameToken (without knowing the processID). 

Happy to change the way this works if you have a suggestion.

Reply:
Well, I think this method doesn't have any layer violation.
However, using BraveTalkService here is layer violation.